pkg/proxy/apis/kubeproxyconfig/validation/validation.go

```diff
@@ -146,9 +146,10 @@ func validateProxyMode(mode kubeproxyconfig.ProxyMode, fldPath *field.Path) fiel
 	case kubeproxyconfig.ProxyModeUserspace:
 	case kubeproxyconfig.ProxyModeIPTables:
 	case kubeproxyconfig.ProxyModeIPVS:
+	case kubeproxyconfig.ProxyModeKernelspace:
 	case "":
 	default:
-		modes := []string{string(kubeproxyconfig.ProxyModeUserspace), string(kubeproxyconfig.ProxyModeIPTables), string(kubeproxyconfig.ProxyModeIPVS)}
+		modes := []string{string(kubeproxyconfig.ProxyModeUserspace), string(kubeproxyconfig.ProxyModeIPTables), string(kubeproxyconfig.ProxyModeIPVS), string(kubeproxyconfig.ProxyModeKernelspace)}
```

Comment:
This logic really needs to validate based on GOOS.  It doesn't make sense to allow options from the wrong OS.

Reply:
Agreed and added. Not sure if you noticed my comment, but should this go into `release-1.9` instead? This is a full-stop blocker for deploying Windows nodes so we'd want this in ASAP.